### PR TITLE
Refactor Courses Page to Use React

### PR DESCRIPTION
**Pull request by varun-iyer**
This pull request refactors the courses page to use react, adding a course's school and term to the page in the process.

Acceptance Critera:
- [ ] All non `Hidden` courses are visible in /courses
- [ ] /courses primarily uses React to display content
- [ ] /courses is searchable and sortable
- [ ] Table in /courses includes Term and School information

- Removed has_many so schools_table is standalone
- Schema commit.
- Re-added has_many relationship
- Created json api for courses
- Created CoursesIndex React, added to Courses index
- pc/vi - WIP on converting courses page to React Bootstrap Table
- Add colors + confirmation to button
- Removed unused CoursesIndexItem

### CODE REVIEW

**Comment by varun-iyer** [comment]
Addresses issue #333, #329, and possibly more.

**Comment by varun-iyer** [comment]
I'm unable to replicate this error on my own machine; investigating.

File: app/javascript/components/Courses/CoursesIndex.jsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+			// 						<th scope="col">Hidden</th>
+			// 						<th scope="col">GitHub Organization</th>
+			// 						<th scope="col" colSpan="2">Actions</th>
+			// 					</tr>
+			// 				</thead>
+			// 				<tbody>
+			// 					{this.state.courses.map(course =>
+			// 						<CoursesIndexItem course={course} onDelete={this.deleteCourse} key={course.id} />
+			// 					)}
+			// 				</tbody>
```
Comment (pconrad) [inline app/javascript/components/Courses/CoursesIndex.jsx:124]: Varun:

Please consider removing this commented out code before we merge this into the master branch

**Review by pconrad — CHANGES_REQUESTED** [review]

File: app/javascript/components/Courses/CoursesIndex.jsx (hunk 1 above)
Comment (varun-iyer) [reply app/javascript/components/Courses/CoursesIndex.jsx]: I believe this change has been pushed by you, thank you for doing so. My apologies for not catching it.

**Review by pconrad — APPROVED** [review]

**Pull request merged**